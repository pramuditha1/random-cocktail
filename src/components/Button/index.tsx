import * as React from "react";
import Button, { ButtonProps } from "@mui/material/Button";

interface PropTypes {
  variant: ButtonProps["variant"];
  text: string;
  onClickHandler: ButtonProps["onClick"];
}

const ButtonComp: React.FC<PropTypes> = ({ variant, text, onClickHandler }) => {
  return (
    <Button variant={variant} onClick={onClickHandler}>
      {text}
    </Button>
  );
};

export default ButtonComp;
