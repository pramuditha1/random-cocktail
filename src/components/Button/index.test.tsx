import React from "react";
import { render } from "@testing-library/react";
import { ButtonProps } from "@mui/material/Button";
import ButtonComp from "./index";

type VariantType = ButtonProps["variant"];
test("ButtonComp matches snapshot", () => {
  const mockProps: {
    variant: VariantType;
    text: string;
    onClickHandler: ButtonProps["onClick"];
  } = {
    variant: "contained",
    text: "Favourite Items",
    onClickHandler: jest.fn(),
  };

  const { asFragment } = render(<ButtonComp {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
