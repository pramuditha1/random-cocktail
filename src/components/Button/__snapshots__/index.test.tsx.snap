// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComp matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Favourite Items
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
